builder/docker/builder.go

```diff
@@ -67,23 +80,18 @@ func (b *Builder) Run(ctx context.Context, ui packer.Ui, hook packer.Hook) (pack
 		log.Print("[DEBUG] Container will be discarded")
 	} else if b.config.Commit {
 		log.Print("[DEBUG] Container will be committed")
-		steps = append(steps, new(StepCommit))
+		steps = append(steps,
+			new(StepCommit),
+			&StepSetGeneratedData{ // Adds ImageSha256 variable available after StepCommit
```

Comment:
@nywilken updated the code to make it easier to see when the variable the available